judgels-frontends/raphael/src/routes/uriel/contests/single/problems/single/ContestProblemPage/ContestProblemPage.tsx

```diff
@@ -7,35 +7,49 @@ import { ContentCard } from 'components/ContentCard/ContentCard';
 import { Contest } from 'modules/api/uriel/contest';
-import { ContestProblem, ContestProblemWorksheet } from 'modules/api/uriel/contestProblem';
-
+import { ContestProblem } from 'modules/api/uriel/contestProblem';
+import { ContestProblemWorksheet as ContestBundleProblemWorksheet } from 'modules/api/uriel/contestProblemBundle';
+import { ContestProblemWorksheet as ContestProgrammingProblemWorksheet } from 'modules/api/uriel/contestProblemProgramming';
+import { ProblemSubmissionFormData as ProgrammingProblemSubmissionFormData } from 'components/ProblemWorksheetCard/Programming/ProblemSubmissionForm/ProblemSubmissionForm';
+import { ProblemWorksheet as ProgrammingProblemWorksheet } from 'modules/api/sandalphon/problemProgramming';
+import { ProblemWorksheet as BundleProblemWorksheet } from 'modules/api/sandalphon/problemBundle';
 import { selectContest } from '../../../../modules/contestSelectors';
 import { contestProblemActions as injectedContestProblemActions } from '../../modules/contestProblemActions';
-import { contestSubmissionActions as injectedContestSubmissionActions } from '../../../submissions/modules/contestSubmissionActions';
+import { contestProgrammingSubmissionActions as injectedContestProgrammingSubmissionActions } from '../../../submissions/modules/contestProgrammingSubmissionActions';
+import { contestBundleSubmissionActions as injectedContestBundleSubmissionActions } from '../../../submissions/modules/contestBundleSubmissionActions';
 import { breadcrumbsActions as injectedBreadcrumbsActions } from 'modules/breadcrumbs/breadcrumbsActions';
+import { ProblemWorksheetCard as ProgrammingProblemWorksheetCard } from 'components/ProblemWorksheetCard/Programming/ProblemWorksheetCard';
+import { ProblemWorksheetCard as BundleProblemWorksheetCard } from 'components/ProblemWorksheetCard/Bundle/ProblemWorksheetCard';
 
 import './ContestProblemPage.css';
+import { ItemSubmission } from 'modules/api/sandalphon/submissionBundle';
 
 export interface ContestProblemPageProps extends RouteComponentProps<{ problemAlias: string }> {
   contest: Contest;
   statementLanguage: string;
-  onGetProblemWorksheet: (
+  onGetProblemInfo: (contestJid: string, problemAlias: string) => Promise<ProblemInfo>;
+  onGetProgrammingProblemWorksheet: (
     contestJid: string,
     problemAlias: string,
     language?: string
-  ) => Promise<ContestProblemWorksheet>;
-  onCreateSubmission: (
+  ) => Promise<ContestProgrammingProblemWorksheet>;
+  onGetBundleProblemWorksheet: (
+    contestJid: string,
+    problemAlias: string,
+    language?: string
+  ) => Promise<ContestBundleProblemWorksheet>;
+  onCreateProgrammingSubmission: (
     contestJid: string,
     contestSlug: string,
     problemJid: string,
-    data: ProblemSubmissionFormData
+    data: ProgrammingProblemSubmissionFormData
   ) => Promise<void>;
+  onCreateBundleSubmission: (contestJid: string, problemJid: string, itemJid: string, answer?: string) => Promise<void>;
```

Comment:
```suggestion
  onCreateBundleSubmission: (contestJid: string, problemJid: string, itemJid: string, answer: string) => Promise<void>;
```

`answer` is always passed